lib/autoupdate/agent/telemetry.go

```diff
@@ -103,3 +107,36 @@ func hasParentDir(dir, parent string) (bool, error) {
 	}
 	return strings.HasPrefix(absDir, absParent), nil
 }
+
+// ReadHelloUpdaterInfo reads the updater config and generates a proto.UpdaterV2Info
+// that can be reported in the inventory hello message.
+// This function performs io operations, its usage must be cached
+// (the downstream inventory handler does this for us).
+func ReadHelloUpdaterInfo() (*proto.UpdaterV2Info, error) {
+	info := &proto.UpdaterV2Info{}
+
+	configPath := os.Getenv(updateConfigPathEnvVar)
+	if configPath == "" {
+		return nil, trace.NotFound("config file %q not found", configPath)
+	}
+
+	conf, err := readConfig(configPath)
+	if err != nil {
+		return nil, trace.Wrap(err, "reading config file %q", configPath)
+	}
+
+	info.UpdateGroup = conf.Spec.Group
+
+	// TODO(hugoShaka or sclevine): After the new UUId logic, add the UUID export here
+
+	switch {
+	case conf.Spec.Enabled == false:
+		info.UpdaterStatus = proto.UpdaterStatus_UPDATER_STATUS_DISABLED
+	case conf.Spec.Pinned == true:
+		info.UpdaterStatus = proto.UpdaterStatus_UPDATER_STATUS_PINNED
```

Comment:
```suggestion
	switch {
	case !conf.Spec.Enabled:
		info.UpdaterStatus = proto.UpdaterStatus_UPDATER_STATUS_DISABLED
	case conf.Spec.Pinned:
		info.UpdaterStatus = proto.UpdaterStatus_UPDATER_STATUS_PINNED
```
(already adjusted on my branch, so feel free leave as-is)